### PR TITLE
Goal GET request

### PR DESCRIPTION
**Pull request by MichaelPHanks**
Added GET request for all Goal items in the database.

### CODE REVIEW

**Review by calebedwards21 — DISMISSED** [review]

File: backend/internal/models/goals.go [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+	UserId   			uint          	`json:"user_id"`
+	GameId   			uint          	`json:"game_id"` 
+	//Game     			Game          	`gorm:"game"` TODO: When seeding is finished we can officially test this
+	TeamId   			uint          	`json:"team_id"` 
+	//Team     			Team			`gorm:"team"` TODO: When seeding is finished we can officially test this
+	Duration 			uint			`json:"duration"` // Do we potentially want to change this to time.Duration
+	Period	    		uint			`json:"period"`
+	Assist1Id			uint   	   		`json:"assist1_id"`
+	Assist2Id			uint   	   		`json:"assist2_id"`
+	PlayerDifferential	int		   		`json:"playerdifferential"`
```
Comment (ryanbarlow1) [inline backend/internal/models/goals.go:16]: Is there a reason why this is an int while the others are uint?

File: backend/internal/server/apis/stats/goals.go [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+
+
+func init() {
+	apis.RegisterHandler(fiber.MethodPost, "/goals", auth.Public, postGoalsHandler)
+	apis.RegisterHandler(fiber.MethodGet, "/goals", auth.Public, getGoalsHandler)
+}
+
+func postGoalsHandler(c *fiber.Ctx) error {
+	log := locals.Logger(c)
+	log.Info("Handling creating new goal")
```
Comment (ryanbarlow1) [inline backend/internal/server/apis/stats/goals.go]: We don't need this first log.Info, Dr. Christensen said the logger does this automatically

File: backend/internal/server/apis/stats/goals.go [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+		return responder.BadRequest(c, "Could not post goal into database")
+	}
+	
+	return responder.Ok(c)
+
+}
+
+func getGoalsHandler(c *fiber.Ctx) error {
+	log := locals.Logger(c)
+	log.Info("Handling getting all goals")
```
Comment (ryanbarlow1) [inline backend/internal/server/apis/stats/goals.go]: We don't need this first log.Info, Dr. Christensen said the logger does this automatically

File: backend/internal/server/apis/stats/goals.go [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+	jsonData, err := json.Marshal(goals)
+	if err != nil {
+		log.WithErr(err).Alert("Failed to serialize goals response payload")
+		return err
+	}
+
+	c.Type("json")
+
+	// Send JSON response
+	return c.Send(jsonData)
```
Comment (ryanbarlow1) [inline backend/internal/server/apis/stats/goals.go]: [Here's an example](https://github.com/tannerwheeler/powerplay-backend-stats/blob/penalty-improvements/backend/internal/server/apis/stats/penalty.go) of how to do this using the `responder.OkWithData()` that Dr. Christensen mentioned we should be using, I just updated the Penalty API in a branch yesterday.

File: backend/internal/models/goals.go (hunk 1 above)
Comment (MichaelPHanks) [reply backend/internal/models/goals.go:16]: note to other pull request

File: backend/internal/server/apis/stats/goals.go (hunk 2 above)
Comment (MichaelPHanks) [reply backend/internal/server/apis/stats/goals.go]: fixed in other pull request

File: backend/internal/server/apis/stats/goals.go (hunk 3 above)
Comment (MichaelPHanks) [reply backend/internal/server/apis/stats/goals.go]: fixed

File: backend/internal/server/apis/stats/goals.go (hunk 4 above)
Comment (MichaelPHanks) [reply backend/internal/server/apis/stats/goals.go]: fixed

**Review by tannerwheeler — CHANGES_REQUESTED** [review]
image
I am unable to test the Get Request. The image shows what I get when I load localhost:9001/docs
Please let me know if I am doing something incorrect or if you get something different.

**Review by tannerwheeler — APPROVED** [review]
Nevermind the cache did not clear. I was able to get the successful POST and GET. Great Job!

**Review by ryanbarlow1 — APPROVED** [review]

**Pull request merged**